xarray/core/indexing.py

```diff
@@ -654,6 +692,204 @@ def _outer_to_numpy_indexer(key, shape):
         return _outer_to_vectorized_indexer(key, shape)
 
 
+class IndexingSupport(object):  # could inherit from enum.Enum on Python 3
+    # for backends that support only basic indexer
+    BASIC = 'BASIC'
+    # for backends that support basic / outer indexer
+    OUTER = 'OUTER'
+    # for backends that support outer indexer including at most 1 vector.
+    OUTER_1VECTOR = 'OUTER_1VECTOR'
+    # for backends that support full vectorized indexer.
+    VECTORIZED = 'VECTORIZED'
```

Comment:
I was just thinking. Is there a page some where that neatly and clearly documents the terms "basic", "outer", and "vectorized" indexing? I honestly keep getting myself mixed up what is what, and it would be good to have a single spot that explains the difference, and have that page referenced in difference places throughout the codebase, like here.

Reply:
https://github.com/numpy/numpy/pull/6256 is probably the closest. When I have time, I want to finish that up and merge it....